### DISCUSSION TITLE
Computed properties are cached… Are they?

### DISCUSSION THREAD

**Discussion by mityukov** [livewire/livewire#3164] [1. Help]
> Computed properties are cached for an individual Livewire request lifecycle. Meaning, if you call `$this->post` 5 times in a component's blade view, it won't make a separate database query every time.

Hi, need to know this for sure, because I put some heavy lifting into those methods, relying on this caching. But if I log accesses to this method — there are plenty of them (within one request)… Is it expected behavior (like, "when there is a sideeffect — it's not cached by design")? Or, is there something else I should check?

**Comment by mityukov** [comment]
Here's the code:
```php
public function getWidgetActivityProperty()
{
        info(__METHOD__, [$this->isWidget, $this->leadId]);

        if (! $this->isWidget || ! $this->leadId) {
            return null;
        }

        return $this->getWidgetActivity();
 }
```


btw, it looks it's only repeated if the method returns `null`. If the calculation result  is not null — it won't repeat. Is it by design?

**Comment by joshhanley** [comment]
@mityukov yeah computed properties are memoized after they have been computed, but that is confined to a single request lifecycle. So on subsequent requests it would recalculate.

The way the memoization is currently setup, it does a check for `isset` on the computer property cache, so as you mentioned, returning null from the computed property will mean that check fails, so it won't use a memoized value and instead run again.

You can see the code for that here https://github.com/livewire/livewire/blob/06f7c7681aaa447ddaa9c26c8a27cf8b8826c15c/src/Component.php#L222-L228

Hope this helps!

**Reply by calebporzio** [reply]
Ah interesting - if there is an easy way to always cache even if it returns null, we should do that. Something like array_key_exists instead of isset() might do it?

I'd be up for a PR of this.

**Reply by joshhanley** [reply]
@calebporzio yeah agreed, array_key_exists would be a better option here
